### PR TITLE
Remove reset buttons from viz settings

### LINKED ISSUE #13955: We should remove all the reset buttons littering the visualization settings
Somewhere along the way, Tom talked me into the idea of adding the little reset arrow button to almost every visualization setting. But it borders on the absurd now and adds unnecessary clutter and confusion. We should get rid of them.

There could be some cases where it's possible currently to change your visualization so much that you can't get it back to its original state — but if that's the case, that's a separate UX problem that I'd like to solve on a case by case basis.

So having set that, I can't really think of a reason why we should keep these individual reset buttons. If we want, we could consider having a single global reset button for the whole viz, like how we do with dashboard card viz settings.

image

### PR DESCRIPTION
**Pull request by mazameli**
Resolves https://github.com/metabase/metabase/issues/13955

Every time you change most visualization settings, a dumb reset button appears next to its label. I can't think of a reason why we should keep these. Are there any legitimate cases I'm forgetting or missing?

## Before

image

## After

image

### CODE REVIEW

**Comment by flamber** [comment]
I use it constantly as it gives an indication if a setting differs from it's parent, and make it possible to reset a setting and then I know it will use the parents setting. I really miss the "Reset to defaults" since 0.33 too.

**Review by kdoh — COMMENTED** [review]
The diff itself looks ok and I'm especially in favor of removing this if the setting is a toggle since having the reset there is just straight up silly. Before we go all the way though per @flamber's comment I do wonder if there's a case to be made that we should maybe have a "reset" for the whole viz type? Or are there other settings where the default value is harder to remember where we might want to keep a "clear" style action.

File: frontend/src/metabase/visualizations/components/ChartSettingsWidget.jsx [hunk 1]
```diff
@@ -32,20 +30,7 @@ const ChartSettingsWidget = ({
         disable: disabled,
       })}
     >
-      {title && (
-        <h4 className="mb1 flex align-center">
-          {title}
-          <Icon
-            size={12}
-            className={cx("ml1 text-light text-medium-hover cursor-pointer", {
-              hidden: !set || noReset,
```
Comment (paulrosenzweig) [inline frontend/src/metabase/visualizations/components/ChartSettingsWidget.jsx:41]: It looks like `set` and `noReset` aren't needed in this component anymore. They could at least be dropped from the destructured props and I'm guessing `noReset` can be deleted upstream too.

Comment (mazameli) [reply frontend/src/metabase/visualizations/components/ChartSettingsWidget.jsx:41]: Thanks @paulrosenzweig. I've attempted doing that in [this latest commit](https://github.com/metabase/metabase/pull/13968/commits/4355ad5a23a0235365b062aa802f82cc5bd6355d) — does it seem right?

**Review by paulrosenzweig — APPROVED** [review]
Looks good!

**Comment by mazameli** [comment]
Okay, I'm merging this. I think we can add a reset for the whole viz if we get feedback to that effect.

@flamber's feedback primarily had to do with these buttons being helpful from a perspective of trying to understand/debug global vs. column-level formatting settings. I accept that as a real problem, but my opinion is that reveals an upstream UX issue that should be addressed more directly.

**Pull request merged**